crates/futures/src/task/singlethread.rs

```diff
@@ -19,13 +19,16 @@ pub(crate) struct Task {
 
     // This is used to ensure that the Task will only be queued once
     is_queued: Cell<bool>,
+    // Ensure that a task will be delayed to the next tick on first run
+    first_run: Cell<bool>,
```

Comment:
I don't think that this is necessary to store here since the first `wake_by_ref` could probably just inline the queue management or call a different helper function with the argument passed through?